ui/src/app/base/sagas/actions.js

```diff
@@ -0,0 +1,47 @@
+import { call } from "redux-saga/effects";
+
+import {
+  machine as machineActions,
+  resourcepool as resourcePoolActions,
+} from "app/base/actions";
+
+/**
+ * Generate functions that will use the response to create the dispatchable
+ * action to set the pool for each machine.
+ * @param {Array} machines - A list of machine ids.
+ * @returns {Array} The list of action creator functions.
+ */
+export const generateActionCreators = (machines) =>
```

Comment:
Will every one of these chained websocket sagas need its own function generator like this?

Reply:
Yeah, something like that. It depends on the action that needs to be performed. The generator is outside. This is only external to the saga to make the saga testable.